partiql-planner/src/main/kotlin/org/partiql/planner/metadata/Metadata.kt

```diff
@@ -0,0 +1,12 @@
+package org.partiql.planner.metadata
+
+/**
+ * Top-level metadata interface for access to object descriptors.
```

Comment:
Slightly confused on the distinction between `Metadata` and `Catalog`. Here we define the `Metadata` interface to be "Top-level metadata interface for access to object descriptors."

while in `Namespace.kt`, it's mentioned "top-level namespaces are called "Catalogs"".

Is there a distinction between the two? If not, then could we utilize the same terminology?